addons/survey/models/survey_question.py

```diff
@@ -264,6 +265,36 @@ def _compute_save_as_email(self):
             if question.question_type != 'char_box' or not question.validation_email:
                 question.save_as_email = False
 
+    @api.depends('survey_id.session_speed_rating_time_limit', 'survey_id.session_speed_rating', 'is_time_limited')
+    def _compute_time_limit(self):
+        """If time limit value is set manually on the question, we do not update it
+        when changing the time limit value of the survey or disabling it.
+        """
+        for question in self:
+            if not question.is_time_limited:
+                question.time_limit = False
+            if not question.survey_id.session_speed_rating:
+                if question.time_limit == question.survey_id.session_speed_rating_time_limit:
+                    question.time_limit = False
+                    question.is_time_limited = False
+            elif question.is_time_limited:
+                if question.time_limit == 0 or question.time_limit == question.old_time_limit:
+                    question.time_limit = question.survey_id.session_speed_rating_time_limit or 0
+            else:
+                if question.old_time_limit == 0:
+                    question.time_limit = question.survey_id.session_speed_rating_time_limit or 0
+                    question.is_time_limited = True
+                else:
+                    question.time_limit = False
+                    question.is_time_limited = False
+            if not question.survey_id.session_speed_rating:
+                question.old_time_limit = False
+            else:
+                question.old_time_limit = question.survey_id.session_speed_rating_time_limit or False
+
+    def _inverse_time_limit(self):
```

Comment:
Why is it needed ? It is a computed stored, so either you make is editable (readonly=False),either there is something I did not understand...